code/modules/mining/equipment/mining_charges.dm

```diff
@@ -0,0 +1,210 @@
+/obj/item/grenade/plastic/miningcharge
+	name = "industrial mining charge"
+	desc = "Used to make big holes in rocks. Only works on rocks!"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "mining-charge-2"
+	det_time = 5
+	notify_admins = FALSE // no need to make adminlogs on lavaland, while they are "safe" to use
+	var/timer_off = FALSE
+	var/installed = FALSE
+	var/smoke_amount = 3
+	var/boom_sizes = list(2,3,5)
+	var/hacked = FALSE
+
+/obj/item/grenade/plastic/miningcharge/Initialize()
+	. = ..()
+	image_overlay = mutable_appearance(icon, "[icon_state]_active", ON_EDGED_TURF_LAYER)
+
+/obj/item/grenade/plastic/miningcharge/examine(mob/user)
+	. = ..()
+	if(hacked)
+		. += "Its wiring is haphazardly changed."
+	if(timer_off)
+		. += "<span class='notice'>The mining charge is connected to a detonator.</span>"
+
+/obj/item/grenade/plastic/miningcharge/attack_self(mob/user)
+	if(nadeassembly)
+		nadeassembly.attack_self(user)
+
+/obj/item/grenade/plastic/miningcharge/afterattack(atom/movable/AM, mob/user, flag)
+	if(ismineralturf(AM) || hacked)
+		if(is_ancient_rock(AM) && !hacked)
+			visible_message("<span class='notice'>This rock appears to be resistant to all mining tools except pickaxes!</span>")
+			return
+		if(timer_off) //override original proc for plastic explosions
+			if(!flag)
+				return
+			if(istype(AM, /mob/living/carbon))
+				return
+			to_chat(user, "<span class='notice'>You start planting the [src].</span>")
+			if(do_after(user, 25 * toolspeed, target = AM))
+				if(!user.unEquip(src))
+					return
+				src.target = AM
+				loc = null
+				if(hacked)
+					message_admins("[ADMIN_LOOKUPFLW(user)] planted [src.name] on [target.name] at [ADMIN_COORDJMP(target)]")
+					log_game("planted [name] on [target.name] at [COORD(target)]", user)
+				installed = TRUE
+				target.overlays += image_overlay
+			return
+		..()
```

Comment:
```suggestion
/obj/item/grenade/plastic/miningcharge/afterattack(atom/movable/AM, mob/user, flag)
	if(!ismineralturf(AM) && !hacked)
		return
	if(is_ancient_rock(AM) && !hacked)
		visible_message("<span class='notice'>This rock appears to be resistant to all mining tools except pickaxes!</span>")
		return
	if(timer_off) //override original proc for plastic explosions
		return ..()
	if(!flag)
		return
	if(iscarbon(AM, /mob/living/carbon))
		return
	to_chat(user, "<span class='notice'>You start planting the [src].</span>")
	if(do_after(user, 25 * toolspeed, target = AM))
		if(!user.unEquip(src))
			return
		target = AM
		loc = null
		if(hacked)
			message_admins("[ADMIN_LOOKUPFLW(user)] planted [src.name] on [target.name] at [ADMIN_COORDJMP(target)]")
			log_game("planted [name] on [target.name] at [COORD(target)]", user)
		installed = TRUE
		target.overlays += image_overlay
```
cleaner code, guard clauses

Reply:
That's became a little broken on tests..

Reply:
you can just go back over what I did and manually apply the changes if you want. It's really just formatting and all of that

Reply:
I already fixed it, don't worry. All problems in missing "!"